File: algorithms/1_two_pointers/1_two_sum/two_sum.py
def twoSum(numbers, target):
    pointer_small = 0
    pointer_big = len(numbers)-1
    while pointer_small <= pointer_big:
        if numbers[pointer_small]+numbers[pointer_big] == target:
            return [pointer_small+1, pointer_big+1]
        elif numbers[pointer_small]+numbers[pointer_big] > target:
            pointer_big = pointer_big -1
        else: pointer_small = pointer_small +1

if __name__ == "__main__":
    print(twoSum([2,3,4,5],5))

File: algorithms/1_two_pointers/1_two_sum/two_sum.js
'use strict';

var twoSum = function(numbers, target) {
  let pointer_small = 0;
  let pointer_big = numbers.length-1;
  while ( pointer_small <= pointer_big){
    if (numbers[pointer_small]+numbers[pointer_big]===target) {
      return [pointer_small+1, pointer_big+1];
    }
    else if (numbers[pointer_small]+numbers[pointer_big]>target) {
      pointer_big-= 1;
    }
    else pointer_small+=1;
  }
};

module.exports=twoSum;